read-fonts/src/collections/int_set/mod.rs

```diff
@@ -620,6 +624,56 @@ where
     }
 }
 
+#[cfg(feature = "serde")]
+impl<T: Domain> serde::Serialize for IntSet<T> {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.0.serialize(serializer)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de, T: Domain> serde::Deserialize<'de> for IntSet<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let members = Membership::deserialize(deserializer)?;
+        // we need to verify that the values are valid for the given domain;
+        // we construct a temporary u32 intset so we can iterate the raw values.
+        let temp = IntSet::<u32>(members, PhantomData);
+        // an inverted set checks the domain during iteration, so invalid values
+        // are guarded against.
+        //
+        // HELPME GARRET: please confirm my understanding is correct
```

Comment:
@garretrieger I got tripped up on this and am still not totally confident, let me know if this makes sense to you?

Reply:
What you want to check is that the underlying bitset within the membership enum only contains values that are members of the domain. You'll want to do this in both the inclusive and exclusive (inverted) case.

Reply:
Right okay so I want some way to just iter the bits on the bitset, and check that they're in domain.